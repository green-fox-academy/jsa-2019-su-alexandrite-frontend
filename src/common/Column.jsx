import React from 'react';
import { View, ViewPropTypes } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

const Column = ({ children, style }) => (
  <View style={{ ...styles.column, ...style }}>
    {children}
  </View>
);

Column.propTypes = {
  children: PropTypes.node,
  style: ViewPropTypes.style,
};

Column.defaultProps = {
  children: undefined,
  style: undefined,
};

export default Column;
