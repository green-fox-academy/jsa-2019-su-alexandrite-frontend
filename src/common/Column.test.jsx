import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Column from './Column';

test('Column render properly', () => {
  const tree = renderer.create(
    <Column />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Column render properly with children', () => {
  const tree = renderer.create(
    <Column><Text>test children</Text></Column>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
