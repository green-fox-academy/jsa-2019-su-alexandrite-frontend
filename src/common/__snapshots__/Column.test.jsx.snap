// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column render properly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
/>
`;

exports[`Column render properly with children 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text>
    test children
  </Text>
</View>
`;
